### PR TITLE
DDF-782 Drag and Drop Metacard Ingest Modal

### PR DESCRIPTION
**Pull request by jerenhicks**
Adding a modal to the search UI that will take a file and create a
metacard for it and add it to the system. This modal will contain drag and
drop capabilities, along with selecting the file.

### CODE REVIEW

**Comment by shaundmorris** [comment]
Talked to @menousek about this a bit. Can you please add a way to disable/enable it? I think it would be similar to the ability to sign-in.

**Comment by jerenhicks** [comment]
adding the metatype to control the ingest being shown or not shown.

**Comment by stustison** [comment]
Is this going to include the ability to set incomplete metadata for the metacard that is generated after you drop something into the content endpoint? I know that was part of this ticket, but I don't see that anywhere in the pull request (unless I'm just missing it).

File: search-ui/standard/src/main/webapp/config.js [hunk 1]
```diff
@@ -76,7 +76,10 @@
             pnotify: 'lib/pnotify/jquery.pnotify.min',
 
             // map tools
-            usngs: 'lib/usng/usng'
+            usngs: 'lib/usng/usng',
+
+            //file upload
+            fileupload: 'lib/jquery-file-upload/js/jquery.fileupload'
```
Comment (pklinef) [inline search-ui/standard/src/main/webapp/config.js]: This is jquery related.  Any reason it was not included in the jQuery area above?

File: search-ui/standard/src/main/webapp/js/view/ingest/IngestMenu.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        'js/view/ingest/IngestModal.view',
+        'wreqr'
+    ],
+    function (Marionette, ich, ingestMenuItem, IngestModal, wreqr) {
+
+        if (!ich.ingestMenuItem) {
+            ich.addTemplate('ingestMenuItem', ingestMenuItem);
+        }
+
+        /** Menu.Item = Marionette.Layout.extend({
```
Comment (pklinef) [inline search-ui/standard/src/main/webapp/js/view/ingest/IngestMenu.js]: Did you mean to leave this commented code here?

**Comment by shaundmorris** [comment]
@stustison  you are spot on. I believe we have a ways to get there but that should be the end goal.  Currently we do not have a service that generates metadata and returns it (w/o ingesting it). We could potentially add to the Content REST Endpoint to have an option to return metadata. But open to ideas on that and somebody else can correct if I am wrong. This ticket would be a good way to demonstrate the existing Content Framework functionality of adding files in, metadata being created, and ingested. With the idea of adding the metadata editing/viewing later. My though is we should support both use cases in the end: 1) you drag a file and it inserts into the catalog. 2) you drag a file in, you get some basic metadata, you can edit it, and then it goes into the catalog. Also might need some ticket clarity...

**Comment by mjmeno** [comment]
We should get the ID back of the generated metacard - -then we just have to pull that card and display it in the metacard view.

**Comment by stustison** [comment]
Would we even want to immediately generate a metacard? I was thinking that we'd hand the file to the content framework, it would pull out whatever metadata it could, and then it would hand us back what it found and let us flesh out the rest of the fields. At that point, we could either choose to go forward with creating the record or cancel out of it and not create a metacard in the system.

**Comment by mjmeno** [comment]
That wouldn't be awful, but would require some rearchitecture.  I'd leave that for a future task.  Eventually it'd go into some kind of a "triage" area until the user saved it.

**Comment by shaundmorris** [comment]
:+1:  Think we are all agreed on the functionality of this particularly PR and then what still needs to be done for the future.

File: search-ui/standard/src/main/webapp/config.js (hunk 1 above)
Comment (jerenhicks) [inline search-ui/standard/src/main/webapp/config.js]: moved up to the jquery section

File: search-ui/standard/src/main/webapp/js/view/ingest/IngestMenu.js (hunk 2 above)
Comment (jerenhicks) [inline search-ui/standard/src/main/webapp/js/view/ingest/IngestMenu.js]: nope, was using it as a quick cheat section when I creating the events. Removed.

**Comment by shaundmorris** [comment]
I think all comments have been addressed here. I will merge in COB today if there are no objections.

**Pull request merged**